src/OrchardCore/OrchardCore.Data.YesSql/Removing/ShellDbTablesInfo.cs

```diff
@@ -0,0 +1,331 @@
+    public ISqlDialect Dialect { get; private set; }
+    public ITableNameConvention TableNameConvention { get; private set; }
+    public DbConnection Connection { get; set; }
+    public DbTransaction Transaction { get; set; }
+    public bool ThrowOnError { get; private set; }
+    public ILogger _logger { get; set; } = NullLogger.Instance;
+
+    public HashSet<(string Name, Type Type, string Collection)> MapIndexTables { get; private set; } =
+        new HashSet<(string Name, Type Type, string Collection)>();
+
+    public HashSet<(string Name, Type Type, string Collection)> ReduceIndexTables { get; private set; } =
+        new HashSet<(string Name, Type Type, string Collection)>();
+
+    public HashSet<string> BridgeTables { get; private set; } = new HashSet<string>();
+    public HashSet<string> DocumentTables { get; private set; } = new HashSet<string>();
+    public HashSet<string> Tables { get; private set; } = new HashSet<string>();
+
+    public ShellDbTablesInfo Configure(ShellSettings shellSettings)
+    {
+        DatabaseProvider = shellSettings["DatabaseProvider"];
+
+        return this;
+    }
+
+    public ShellDbTablesInfo Configure(IConfiguration configuration)
+    {
+        Dialect = configuration.SqlDialect;
+        TablePrefix = configuration.TablePrefix;
+        TableNameConvention = configuration.TableNameConvention;
+        _commandInterpreter = configuration.CommandInterpreter;
+
+        return this;
+    }
+
+    public ShellDbTablesInfo Configure(DbTransaction transaction, ILogger logger, bool throwOnError = true)
+    {
+        Transaction = transaction;
+        Connection = transaction.Connection;
+        ThrowOnError = throwOnError;
+
+        return this;
+    }
+
+    public IEnumerable<string> GetTableNames()
+    {
+        return MapIndexTables.Select(i => Prefix(i.Name))
+            .Union(ReduceIndexTables.Select(i => Prefix(i.Name)))
+            .Union(BridgeTables.Select(Prefix))
+            .Append(Prefix(DbBlockIdGenerator.TableName))
+            .Union(DocumentTables.Select(Prefix))
+            .Union(Tables.Select(Prefix))
+            .ToArray();
+    }
+
+    public ISchemaBuilder CreateMapIndexTable(Type indexType, Action<ICreateTableCommand> table, string collection)
+    {
+        var indexTable = TableNameConvention.GetIndexTable(indexType, collection);
+        var documentTable = TableNameConvention.GetDocumentTable(collection);
+
+        MapIndexTables.Add((indexTable, indexType, collection));
+        DocumentTables.Add(documentTable);
+
+        return this;
+    }
+
+    public ISchemaBuilder CreateReduceIndexTable(Type indexType, Action<ICreateTableCommand> table, string collection = null)
+    {
+        var indexTable = TableNameConvention.GetIndexTable(indexType, collection);
+        var documentTable = TableNameConvention.GetDocumentTable(collection);
+        var bridgeTable = indexTable + "_" + documentTable;
```

Comment:
We should not be hard coding `_` in the table names. Each tenant could have a different table separator. You should use `ShellSettings.TablePrefixSeparator` to obtain the correct separator. When `ShellSettings.TablePrefixSeparator` is null, then we use `_`. Note here, we have to use `if(ShellSettings.TablePrefixSeparator is null)` NOT `if(String.IsNullOrEmptrySpace(ShellSettings.TablePrefixSeparator)` as one may want to have no prefix value.

Reply:
Okay will see tomorrow, need to sleep soon.

Did you introduce this option? For now here I'm doing the same as in yesSql

https://github.com/sebastienros/yessql/blob/0b8dd422407b5216882d2dc990d74ab29c4e56cd/src/YesSql.Core/Sql/SchemaBuilder.cs#L102

Reply:
Yes. Actually all you need here is to inject the `ITableNameConvention` and use it to generate the table name as we do in the DbConnectionValidaotor

Reply:
Hmm I think I remember, in which case we don't need all of this.

Here we use configuration coming from YesSql Configuration so for example the TablePrefix already include the separator e.g. "tenant1_". And for example here to generate the bridgeTable, this is what YesSql currently does.

Need to leave, will check it toonight.

Reply:
Here it is not related to `TablePrefix` and `TablePrefixSeparator`, it is a separator between 2 tables as YesSql does.

Reply:
@jtkech I got it. Then it should be using `TableNameConvention` instead of directly hoarding `_` I think this change should also happen in YesSql. In OC, we allow the use to define their own `TableNameConvention` so they may not be using `_` so this would break YesSql and also your code. If you agree, I can submit a PR into YesSql for this change, but for the sake of your PR, I think you should resolve `ITableNameConvention` and let it determine the table name.

Reply:
Yes but only if it is needed and done on YesSql side.

Here we already use `ITableNameConvention` to provide table names, here it is a separator between 2 table names, so would need to be a different configurable separator because the current `TablePrefixSeparator` as it is named is used to separate the `TablePrefix` in a given table name.

Reply:
I think for consistency the  `ITableNameConvention` should be used. I agree with you this would have to be done in YesSql. Let's see what @sebastienros thinks about this. There is probably a good reason why YesSql isn't using `ITableNameConvention` in the SchemaBuilder.

Reply:
Okay no problem.

> why YesSql isn't using ITableNameConvention in the SchemaBuilder.

It uses it, see above line 88 and 89, but not to separate 2 table names to generate the bridge.

Reply:
yes which is why I said maybe there is a valid reason behind not using ` ITableNameConvention` in this case.